aws_lambda_builders/workflows/nodejs_npm/workflow.py

```diff
@@ -213,3 +208,34 @@ def get_resolvers(self):
         specialized path resolver that just returns the list of executable for the runtime on the path.
         """
         return [PathResolver(runtime=self.runtime, binary="npm")]
+
+    @staticmethod
+    def get_install_action(source_dir, artifacts_dir, subprocess_npm, osutils, is_production=True):
+        """
+        Get the install action used to install dependencies at artifacts_dir
+
+        :type source_dir: str
+        :param source_dir: an existing (readable) directory containing source files
+
+        :type artifacts_dir: str
+        :param artifacts_dir: Dependencies will be installed in this directory.
+
+        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
+        :param osutils: An instance of OS Utilities for file manipulation
+
+        :type subprocess_npm: aws_lambda_builders.workflows.nodejs_npm.npm.SubprocessNpm
+        :param subprocess_npm: An instance of the NPM process wrapper
+
+        :type is_production: bool
+        :param is_production: NPM installation mode is production (eg --production=false to force dev dependencies)
+
+        :rtype: BaseAction
+        :return: Install action to use
+        """
+        lockfile_path = osutils.joinpath(source_dir, "package-lock.json")
+        shrinkwrap_path = osutils.joinpath(source_dir, "npm-shrinkwrap.json")
+
+        if osutils.file_exists(lockfile_path) or osutils.file_exists(shrinkwrap_path):
```

Comment:
Question - do we want to check if `npm ci` is available? `npm ci` is available since npm 6.x and nodejs12 comes with npm6.9+. nodejs10 is going to be deprecated in Lambda pretty soon. It seems safe but we might want to confirm.

Reply:
Looks like we already have a PR out to remove Node10 support #319 and Lambda is dropping all support very soon too (Feb 14) https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html. We could potentially add a condition to check the npm version, fallback to `npm install` if needed and remove it after that date.